### PR TITLE
Update machine-controller to v1.46.0

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
Signed-off-by: Waleed Malik <ahmedwaleedmalik@gmail.com>

**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

1. machine-controller updated to v1.46.0
2. `initial_machinedeployment_spotinstances` removed in favor of `initial_machinedeployment_spotinstances_max_price`

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
* machine-controller updated to v1.46.0
* For AWS, variable 'initial_machinedeployment_spotinstances_max_price' is introduced. When set, spot instances will be used for initial machine deployments
```

### CODE REVIEW

File: examples/terraform/aws/output.tf [hunk 1]
```diff
@@ -112,7 +112,11 @@ output "kubeone_workers" {
           diskType         = "gp2"
           ## Only applicable if diskType = io1
           diskIops           = 500
-          isSpotInstance     = var.initial_machinedeployment_spotinstances
+          isSpotInstance     = local.initial_machinedeployment_spotinstances
+          ## Only applicable if isSpotInstance is true
+          spotInstanceConfig = {
+            maxPrice = "var.initial_machinedeployment_spotinstances_max_price"
```
Comment (kron4eg) [inline examples/terraform/aws/output.tf:118]:
```suggestion
            maxPrice = var.initial_machinedeployment_spotinstances_max_price
```

Comment (ahmedwaleedmalik) [reply examples/terraform/aws/output.tf:118]: It expects a string, so this would fail

Comment (kron4eg) [reply examples/terraform/aws/output.tf:118]: then use `format("%d")`, a string `"var.initial_machinedeployment_spotinstances_max_price"` doesn't represent a price, it's just a string.

Comment (ahmedwaleedmalik) [reply examples/terraform/aws/output.tf]: Updated, PTAL.

**Review by kron4eg — APPROVED** [review]

**Pull request merged**